src/libraries/System.Net.HttpListener/tests/TrimmingTests/System.Net.HttpListener.TrimmingTests.proj

```diff
@@ -4,12 +4,15 @@
   <ItemGroup>
     <TestConsoleAppSourceFiles Include="CookieExtensionsTest.Clone.cs">
       <AdditionalSourceFiles>CookieExtensionsTest.Helper.cs</AdditionalSourceFiles>
+      <SkipOnTestRuntimes>browser-wasm</SkipOnTestRuntimes> <!-- Justification: HttpListener.ctor() throws PNSE on wasm -->
```

Comment:
Should this property instead be set in the project itself? How common is it that we could have a source file that is not skipped?

Reply:
It will really depend on the project, in this particular case HttpListener is not supported on wasm, so most likely all of its tests will need to be skipped, but we do have other trimming test projects where only some of them will be disabled but not all, like System.Runtime.TrimmingTests. For projects like HttpListener where all tests will be skipped in a platform, I can instead just condition the ItemGroup to check for TargetOS and TargetArchitecture instead.

Reply:
Or we could have a property global to. the project meaning skip all source files in these runtimes, or metadata which is honored if the project property is not set